src/extension.ts

```diff
@@ -73,18 +73,29 @@ async function startServer(tomcatController: TomcatController, tomcatItem?: Tomc
     }
 }
 
-async function restartServer(tomcatController: TomcatController, tomcatItem?: TomcatServer): Promise<void> {
-    await tomcatController.restartServer(tomcatItem);
+async function restartServer(tomcatController: TomcatController, tomcatItem: TomcatServer): Promise<void> {
+    const server: TomcatServer = await selectServer(tomcatController, tomcatItem);
+    if (!server) {
+        vscode.window.showInformationMessage(DialogMessage.noServer);
+        return;
+    }
+    if (server.isStarted()) {
+        await tomcatController.stopOrRestartServer(server, true);
+    } else {
+        await tomcatController.startServer(server);
+    }
 }
 
 async function stopServer(tomcatController: TomcatController, tomcatItem?: TomcatServer): Promise<void> {
     const server: TomcatServer = await selectServer(tomcatController, tomcatItem);
-    if (server) {
-        if (!server.isStarted()) {
-            vscode.window.showInformationMessage(DialogMessage.serverStopped);
-            return;
-        }
-        await tomcatController.stopServer(server);
+    if (!server) {
```

Comment:
looks like we can extract a common method here since most part is same as `restartServer()`